clients/da-vinci-client/src/test/java/com/linkedin/davinci/helix/VeniceLeaderFollowerStateModelTest.java

```diff
@@ -129,4 +130,25 @@ public void testWhenBecomeOfflineFromStandbyWithVersionDeletion() {
     testStateModel.onBecomeOfflineFromStandby(mockMessage, mockContext);
     verify(spyHeartbeatMonitoringService).removeLagMonitor(any(), eq(testPartition));
   }
+
+  @Test
+  public void testTransitionToDropIsBlockedOnDropPartitionAction() {
+    // if the resource is not the current serving version, state transition thread will not be blocked. However, we
+    // still block it once to wait for drop partition action if it's done asynchronously via SIT
+    CompletableFuture mockDropPartitionFuture = mock(CompletableFuture.class);
+    when(mockDropPartitionFuture.isDone()).thenReturn(false);
+    when(mockIngestionBackend.dropStoragePartitionGracefully(any(), anyInt(), anyInt()))
+        .thenReturn(mockDropPartitionFuture);
+    when(mockStore.getCurrentVersion()).thenReturn(2);
+    testStateModel.onBecomeDroppedFromOffline(mockMessage, mockContext);
+    verify(mockParticipantStateTransitionStats, times(1)).incrementThreadBlockedOnOfflineToDroppedTransitionCount();
+    verify(mockParticipantStateTransitionStats, times(1)).decrementThreadBlockedOnOfflineToDroppedTransitionCount();
+
+    // if the resource is the current serving version, state transition thread will be blocked. And it will be blocked
+    // again for drop partition action if it's done asynchronously via SIT
+    when(mockStore.getCurrentVersion()).thenReturn(1);
+    testStateModel.onBecomeDroppedFromOffline(mockMessage, mockContext);
+    verify(mockParticipantStateTransitionStats, times(3)).incrementThreadBlockedOnOfflineToDroppedTransitionCount();
```

Comment:
Why does this get invoked 3 times?

I can see once [here](https://github.com/linkedin/venice/blob/626568143325c3564eaf1362a1fd3b16e7a73df4/clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/LeaderFollowerPartitionStateModel.java#L191-L194), and another when `waitForDropPartition` is true. Where does the third execution happen?

Reply:
Gotcha.

Nit: Can we reset this mock to make it clear that second invocation of `testStateModel.onBecomeDroppedFromOffline` only calls `incrementThreadBlockedOnOfflineToDroppedTransitionCount` twice?